libbeat/paths/paths.go

```diff
@@ -85,7 +85,7 @@ func (paths *Path) InitPaths(cfg *Path) error {
 	}
 
 	// make sure the data path exists
-	err = os.MkdirAll(paths.Data, 0750)
+	err = os.MkdirAll(paths.Data, 0770)
```

Comment:
I am not familiar with the directory/paths code yet, but my only though is if it is possible to be more conservative with the permissions changes since this only affects heartbeat (as far as I know).

Should we make this change (and the umask change in the Dockerfile above) apply only to heartbeat, instead of every beat as it does now?

Reply:
Yes, I would also prefer a more localized change for heartbeat.

Reply:
To clarify, the issue fixed here applies to `heartbeat` running inside a `elastic-agent` container.

Path initialization is peformed by the first beat that kicks off inside `elastic-agent`. AFAIK, there's no way to specify a order. In that case, the alternative would be to move path initialization inside `elastic-agent`, before beats are spawned. 

As for the entrypoint, I've followed the guideline here: https://github.com/elastic/beats/pull/29708. As it stands, all `umask` operations are performed _outside_ libbeat code. With this, we can restrict the change to docker containers without affecting other type of installs (systemd and others). These install methods still have the umask specified in the service configuration template (0027) and will continue to create files with the same permission level as it is today (0750). 

There's also to note, some of the directories that are being generated from inside `elastic-agent` today are attempted with a higher permission level than the default mask allows. Here are a few examples:

````
$ grep mkdirat /tmp/agent.* | grep "state/data" | grep default\"
/tmp/agent.379:mkdirat(AT_FDCWD, "/usr/share/elastic-agent/state/data/logs/default", 0775) = 0
/tmp/agent.379:mkdirat(AT_FDCWD, "/usr/share/elastic-agent/state/data/tmp/default", 0775) = 0
````

If this permission level can be considered a security concern, I'm guessing we shouldn't be relying on an environmental `umask` to filer them out.

Reply:
Thanks for the explanation! I don't have major concerns about this change.

@rdner thoughts on this? You were the last person to touch anything to do with umask.

Reply:
I can confirm that this particular line should not affect other kinds of distribution since they all have their `umask` set in their respective scripts which would overwrite the permission set for this directory.

However, I have some concerns that we now have a different `umask` for all beats running under `elastic-agent` because of the change in the Docker template. If that's okay for everyone I'm okay with that too.

The whole `umask` story started with this security issue which was primarily about the access by `all`, I guess giving more access to the `group` is fine https://github.com/elastic/beats/issues/14005